.github/workflows/publish.yml

```diff
@@ -0,0 +1,52 @@
+name: publish
+on:
+  push:
+    tags:
+      - 'v.*'
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Build package
+        run: |
+          python -m pip install wheel jupyter-packaging jupyterlab>=3
+          BUILD_JUPYTERLAB_EXTENSION=1 python setup.py sdist bdist_wheel
+          # Don't publish a tar.gz file over 1MB  (Issue #730)
+          if (($(wc -c < dist/*.tar.gz) > 1000000)); then exit 1; fi
+          # node_modules should not be in the package
+          if (($(tar -tf dist/*.tar.gz | grep node_modules | wc -l)>0)); then echo "node_modules should not be included" && exit 1; fi
+          # Check that the lab and the notebook extensions are there
+          if (($(tar -tf dist/*.tar.gz | grep packages/labextension/package.json$ | wc -l)==0)); then echo "Missing lab extension" && exit 1; fi
+          if (($(tar -tf dist/*.tar.gz | grep jupytext/nbextension/index.js$ | wc -l)==0)); then echo "Missing notebook extension" && exit 1; fi
+          # Install
+          python -m pip install dist/*.tar.gz
+          echo "Install went OK"
+  publish:
+    name: Publish to PyPi
+    needs: build
+    if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Build package
+        run: |
+          python -m pip install wheel jupyter-packaging jupyterlab>=3
+          BUILD_JUPYTERLAB_EXTENSION=1 python setup.py sdist bdist_wheel
+      - name: Publish
+        uses: pypa/gh-action-pypi-publish@v1.1.0
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_KEY }}
```

Comment:
See https://github.com/pypa/gh-action-pypi-publish#trusted-publishing
Thid is the preferred approach these days